File: Fontes de exemplos/Comunicação com SOCKET/CSharp/ComPort/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.IO.Ports;
using System.Net;
using System.Net.Sockets;
using System.IO;
using System.Collections;
using System.Threading;

namespace ComPort
{
    public partial class Form1 : Form
    {

        string ip;
        TcpClient client;
        NetworkStream stream;
        public Form1()
        {
            InitializeComponent();
        }        

        private void btnSend_Click(object sender, EventArgs e)
        {
            string command = txtCommand.Text;
            Byte[] data = System.Text.Encoding.ASCII.GetBytes(command);
            stream.Write(data, 0, data.Length);
            data = new Byte[256];
            String responseData = String.Empty;            
            Int32 bytes = stream.Read(data, 0, data.Length);
            responseData = System.Text.Encoding.ASCII.GetString(data, 0, bytes);
            txtAnswer.Text = responseData;
        }

        private void btnConect_Click(object sender, EventArgs e)
        {
            String ip = txtIp.Text;
            Int32 port = Convert.ToInt32(txtPort.Text);            

            client = new TcpClient(ip, port);
            stream = client.GetStream();
            ckConected.Checked = true;
        }

        private void btnDisconect_Click(object sender, EventArgs e)
        {
            client.Close();
            stream.Close();
            ckConected.Checked = false;
        }
    }
}

File: concentradores/Fontes de exemplos/Comunicação com SOCKET/VB/TesteSocketVB/Form1.vb
Imports System.Net
Imports System.Net.Sockets
Imports System.Text
Public Class Form1
    Dim client As TcpClient
    Dim stream As NetworkStream
    Private Sub btnConect_Click(sender As Object, e As EventArgs) Handles btnConect.Click
        Dim ip As String = txtIp.Text
        Dim port As Int32 = Convert.ToString(txtPort.Text)
        client = New TcpClient(ip, port)
        stream = client.GetStream()
        ckConected.Checked = True
    End Sub

    Private Sub btnDisconect_Click(sender As Object, e As EventArgs) Handles btnDisconect.Click
        client.Close()
        stream.Close()
        ckConected.Checked = False
    End Sub

    Private Sub btnSend_Click(sender As Object, e As EventArgs) Handles btnSend.Click
        Dim command As String = txtCommand.Text
        Dim data As [Byte]() = Encoding.ASCII.GetBytes(txtCommand.Text)
        stream.Write(data, 0, data.Length)

        Dim teste(126) As Byte
        Dim bytes As Int32 = stream.Read(teste, 0, teste.Length)
        Dim responseData As String = Encoding.ASCII.GetString(teste, 0, bytes)
        txtAnswer.Text = responseData
    End Sub
End Class
